aidies-project/client/src/components/CampaignCreate.js

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { useParams, useHistory } from 'react-router-dom';
+
+import { getOneCampaign, bid, setLocation } from '../store/campaign';
+import { DetermineBid, DetermineTimeRemaining } from './CampaignUtils';
+import { makeStyles } from '@material-ui/core/styles';
+import { Card, Typography, Button} from '@material-ui/core';
+import '../css/campaign.css';
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        '& > *': {
+            margin: theme.spacing(1),
+        },
+    },
+}));
+const CampaignCreate = ({ needLogin }) => {
+    const classes = useStyles();
+    const { id } = useParams();
+    const history = useHistory();
+    const location = useLocation();
+    const campaign = useSelector(state => state.campaign.current);
+    let highest = useSelector(state => state.campaign.highestBid);
+    const userId = useSelector(state => state.authentication.user.id);
+    const [anchorEl, setAnchorEl] = useState(null);
+
+    const dispatch = useDispatch();
+    useEffect(() => {
+        dispatch(getOneCampaign(id))
+    }, [id]);
+
+    if (!campaign || !campaign.User) {
+        return null;
+    }
+
+    const handleBid = () => {
+        if (needLogin) {
+            dispatch(setLocation(location.pathname));
+            return history.push("/login");
+        }
+        if (!highest) {
+            let increaseBid = parseInt(campaign.startingPrice) + 5;
+            dispatch(bid(id, increaseBid, userId));
+
+        } else {
+            let increaseBid = highest + 5;
+            dispatch(bid(id, increaseBid, userId));
+        }
```

Comment:
You can DRY up your code with a refactor like this!
```suggestion
        let increaseBid = parseInt(campaign.startingPrice) + 5;
        if (highest) {
            increaseBid = highest + 5;
        }
        dispatch(bid(id, increaseBid, userId));
```

Reply:
good call